docs/source/integrations/deltalake_versioning.md

```diff
@@ -0,0 +1,107 @@
+
+```bash
+kedro-datasets[pandas-deltatabledataset]
+```
+
+Now, you can install the project dependencies by running:
+
+```bash
+pip install -r requirements.txt
+```
+
+## Using Delta tables in catalog
+
+To use Delta tables in your Kedro project, you can update the `base/catalog.yml` to use `type: pandas.DeltaTableDataset` for the datasets you want to save as Delta tables. For this example, let us update the `model_input_table` dataset in the `base/catalog.yml` file:
+
+```yaml
+model_input_table:
+  type: pandas.DeltaTableDataset
+  filepath: data/02_intermediate/model_input_table
+  save_args:
+    mode: overwrite
+```
+
+You can specify `save_args` to specify the mode of saving the Delta table. The `mode` parameter can "overwrite" or "append" depending on whether you want to overwrite the existing Delta table or append to it. You can also specify [additional saving options that are accepted by the `write_deltalake` function in the `delta-rs` library](https://delta-io.github.io/delta-rs/python/api_reference.html#writing-deltatables) which is used by `pandas.DeltaTableDataset` under the hood.
+
+When you run the Kedro project with `kedro run` command, the Delta table will be saved to the location specified in the `filepath` argument as a folder of `parquet` files. This folder also contains a `_delta_log` directory which stores the transaction log of the Delta table. Subsequent runs of the pipeline will create new versions of the Delta table in the same location and new entries in the `_delta_log` directory.
+
+To load a specific version of the dataset, you can specify the version number in the `load_args` parameter in the catalog entry:
+
+```yaml
+model_input_table:
+  type: pandas.DeltaTableDataset
+  filepath: data/02_intermediate/model_input_table
+  load_args:
+    version: 1
+```
+
+## Inspect the dataset in interactive mode
+
+You can inspect the history and the metadata of the Delta table in an interactive Python session. To start the IPython session with Kedro components loaded, run:
+
+```bash
+kedro ipython
+```
+
+You can load the Delta table using the `catalog.load` method and inspect the dataset:
+
+```python
+model_input_table = catalog.datasets['model_input_table']
+```
+You can inspect the history of the Delta table by accessing the `history` attribute:
+```python
+>> model_input_table.history
+
+[Out]: [
+    {
+        'timestamp': 1739891304488,
+        'operation': 'WRITE',
+        'operationParameters': {'mode': 'Overwrite'},
+        'operationMetrics': {
+            'execution_time_ms': 8,
+            'num_added_files': 1,
+            'num_added_rows': 6027,
+            'num_partitions': 0,
+            'num_removed_files': 1
+        },
+        'clientVersion': 'delta-rs.0.23.1',
+        'version': 1
```

Comment:
I think I followed the steps (I admit I'm reading quickly) and I only have 1 version. Maybe we can explicitly tell the user to make a change and do a `kedro run` again.